pkg/config/config.go

```diff
@@ -0,0 +1,88 @@
+	GitGPGKeyImport          []string      `mapstructure:"git-gpg-key-import"`
+	GitVerifySignatures      bool          `mapstructure:"git-verify-signatures"`
+	GitSigningKey            string        `mapstructure:"git-signing-key"`
+	SyncInterval             time.Duration `mapstructure:"sync-interval"`
+	SyncTimeout              time.Duration `mapstructure:"sync-timeout"`
+	SyncGarbageCollection    bool          `mapstructure:"sync-garbage-collection"`
+	SyncGarbageCollectionDry bool          `mapstructure:"sync-garbage-collection-dry"`
+	SyncState                string        `mapstructure:"sync-state"`
+	MemcachedHostname        string        `mapstructure:"memcached-hostname"`
+	MemcachedPort            int           `mapstructure:"memcached-port"`
+	MemcachedService         string        `mapstructure:"memcached-service"`
+	MemcachedTimeout         time.Duration `mapstructure:"memcached-timeout"`
+	AutomationInterval       time.Duration `mapstructure:"automation-interval"`
+	RegistryPollInterval     time.Duration `mapstructure:"registry-poll-interval"`
+	RegistryRPS              float64       `mapstructure:"registry-rps"`
+	RegistryBurst            int           `mapstructure:"registry-burst"`
+	RegistryTrace            bool          `mapstructure:"registry-trace"`
+	RegistryInsecureHost     []string      `mapstructure:"registry-insecure-host"`
+	RegistryExcludeImage     []string      `mapstructure:"registry-exclude-image"`
+	RegistryUseLabels        []string      `mapstructure:"registry-use-labels"`
+	RegistryECRRegion        []string      `mapstructure:"registry-ecr-region"`
+	RegistryECRIncludeID     []string      `mapstructure:"registry-ecr-include-id"`
+	RegistryECRExcludeID     []string      `mapstructure:"registry-ecr-exclude-id"`
+	RegistryRequire          []string      `mapstructure:"registry-require"`
+	K8sInCluster             bool          `mapstructure:"k8s-in-cluster"`
+	K8sSecretName            string        `mapstructure:"k8s-secret-name"`
+	K8sSecretVolumeMountPath string        `mapstructure:"k8s-secret-volume-mount-path"`
+	K8sSecretDataKey         string        `mapstructure:"k8s-secret-data-key"`
+	K8sNamespaceWhitelist    []string      `mapstructure:"k8s-namespace-whitelist"`
+	K8sAllowNamespace        []string      `mapstructure:"k8s-allow-namespace"`
+	K8sDefaultNamespace      string        `mapstructure:"k8s-default-namespace"`
+	K8sVerbosity             int           `mapstructure:"k8s-verbosity"`
+	SSHKeygenDir             string        `mapstructure:"ssh-keygen-dir"`
+	ManifestGeneration       bool          `mapstructure:"manifest-generation"`
+	Connect                  string        `mapstructure:"connect"`
+	Token                    string        `mapstructure:"token"`
+	RPCTimeout               time.Duration `mapstructure:"rpc-timeout"`
+	DockerConfig             string        `mapstructure:"docker-config"`
+}
+
+func (c Config) IsValid() error {
```

Comment:
Is this invoked implicitly by Viper?

Reply:
No, it's called in main.go: https://github.com/fluxcd/flux/pull/2544/files#diff-350e484490d6241fa81c13942d160480R263

It's a bit of a dance because viper wants to cleverly do everything itself.

Reply:
Sorry, I missed it